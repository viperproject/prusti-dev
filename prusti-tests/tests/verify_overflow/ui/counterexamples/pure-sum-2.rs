// compile-flags: -Punsafe_core_proof=true -Pcounterexample=true -Pcheck_overflows=false

use prusti_contracts::*;

#[pure]
#[terminates(Int::new(x) + Int::new(1))]
fn sum(x:i64) -> i64 {
    if x <= 0 {
        0
    } else {
        x + sum(x - 1)
    }
}

#[ensures(sum(5) == 0)] //TODO: add ce support for pure functions in specifications
fn test1() {}

fn main() {}
