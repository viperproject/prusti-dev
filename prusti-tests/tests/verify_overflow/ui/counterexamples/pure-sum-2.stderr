error: [Prusti: verification error] postcondition might not hold.
  --> $DIR/pure-sum-2.rs:15:11
   |
15 | #[ensures(sum(5) == 0)] //TODO: add ce support for pure functions in specifications
   |           ^^^^^^^^^^^
   |
note: the error originates here
  --> $DIR/pure-sum-2.rs:16:1
   |
16 | fn test1() {}
   | ^^^^^^^^^^^^^
note: counterexample for "result"
       value:   ()
  --> $DIR/pure-sum-2.rs:16:12
   |
16 | fn test1() {}
   |            ^^

error: aborting due to previous error

